apps/wallet/src/engine/runners/solana/collectibles.ts

```diff
@@ -87,22 +93,17 @@ export const loadCollectibleMetadata = async (
 	metadata: Metadata | Nft | Sft,
 	owner: PublicKey,
 ): Promise<SftWithToken | NftWithToken> => {
-	let nft = await mpl.nfts().load({
-		metadata: metadata as Metadata<JsonMetadata<string>>,
-		tokenOwner: owner,
-	});
-
-	if (!metadata.json && 'mintAddress' in metadata) {
-		const nftByMint = await mpl.nfts().findByMint({
-			mintAddress: metadata.mintAddress,
+	const nft = {
+		...(await mpl.nfts().load({
```

Comment:
I guess we don't need to clone this object, as no one else using/reference it

Reply:
actually, the obj return by findByMint is frozen and is read-only